library/core/src/num/flt2dec/strategy/grisu.rs

```diff
@@ -159,6 +159,17 @@ pub fn max_pow10_no_more_than(x: u32) -> (u8, u32) {
     }
 }
 
+/// Given `n`, where `0 <= n <= 9`, returns `10^n`.
+#[doc(hidden)]
+pub fn pow10_up_to_9(n: usize) -> u32 {
+    debug_assert!(n <= 9);
```

Comment:
This can be dropped -- indexing the array gives you a hard assert that n is in `[0, 9)`.

I might just move the array into the function below and index directly as well, seems simpler. This function shouldn't need to be public regardless.